app/components/DataCard.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+import Card from "react-bootstrap/Card";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
+import DataCardInput from "./DataCardInput";
+import { useMutation, useQuery, useLookupQuery } from "@/utils/graphql";
+import {
+  getRecordValue,
+  renderColumnValue,
+  valueToString,
+  handleFormValueOutput,
+} from "@/utils/formHelpers";
+import { ColDataCardDef, LookupTableOption } from "@/types/types";
+
+interface DataCardProps<T extends Record<string, unknown>> {
+  record: T;
+  columns: ColDataCardDef<T>[];
+  mutation: string;
+  isValidating: boolean;
+  title: string;
+  onSaveCallback: () => Promise<void>;
+}
+
+/**
+ * Generic component which renders editable fields in a Card
+ */
+export default function DataCard<T extends Record<string, unknown>>({
```

Comment:
This component replaces the `DataTable` component in VZE.  The only significant difference is that it fetches lookup table values when a lookup field is toggled into its edit state, as opposed to fetching all lookup values when the crash details page loads.

The lookup queries are cached, and in general are quick to load, but we need to test this with our power users, who may notice a slight performance degradation.